### PR TITLE
Add PHP 8 support

### PR DESCRIPTION
**Pull request by daniel-valchev-paysera**
- Added support for PHP 8.
- Upgraded `sentry/sentry-symfony` to ^4.0.
- Removed `paysera/lib-php-cs-fixer-config` as it lacks support for PHP 8.
- Breaking changes in the configuration are described in the `UPGRADE.md` file.

Tested with Sentry 9.1.2, Sentry 20.7.1, Sentry cloud.

### CODE REVIEW

File: src/Monolog/SentryHandler.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Paysera\LoggingExtraBundle\Monolog;
```
Comment (borilyordanov) [inline src/Monolog/SentryHandler.php]: Let's move this to `Paysera\LoggingExtraBundle\Service\Handler`.

File: src/Monolog/SentryHandler.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+use Monolog\Handler\HandlerWrapper;
+use Monolog\Handler\ProcessableHandlerTrait;
+use Sentry\State\Scope;
+use function Sentry\withScope;
+
+final class SentryHandler extends HandlerWrapper
```
Comment (borilyordanov) [inline src/Monolog/SentryHandler.php]: What do you think about `SentryExtraInformationHandler`?

File: tests/Functional/Fixtures/config/cases/basic.yml [hunk 3]
```diff
@@ -14,8 +14,8 @@ monolog:
             channels: ["!event", "!doctrine"]
         sentry:
             type: service
-            id: Sentry\Monolog\Handler
-
+            id: paysera_logging_extra.sentry_handler
+
```
Comment (borilyordanov) [inline tests/Functional/Fixtures/config/cases/basic.yml]: Unnecessary whitespace.

File: UPGRADE-2.0.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# Upgrade 1.x to 2.0
```
Comment (borilyordanov) [inline UPGRADE-2.0.md]: `Upgrade from 1.x to 2.0`

File: src/Resources/config/services.xml [hunk 5]
```diff
@@ -13,5 +13,15 @@
                  class="Paysera\LoggingExtraBundle\Service\CorrelationIdProvider">
             <argument>%paysera_logging_extra.application_name%</argument>
         </service>
+
+        <service id="paysera_logging_extra.sentry_handler"
```
Comment (borilyordanov) [inline src/Resources/config/services.xml]: Let's move this definition under `services/handlers.xml` and rename it to `paysera_logging_extra.handler.sentry`. Then we can define an alias here called `paysera_logging_extra.sentry_handler`, which points to the concrete handler.

**Review by borilyordanov — CHANGES_REQUESTED** [review]

File: composer.json [hunk 6]
```diff
@@ -15,31 +15,27 @@
...
-        "doctrine/doctrine-bundle": "^1.4",
-        "doctrine/orm": "^2.5.14",
-        "doctrine/persistence": "^1.3",
-        "mockery/mockery": "^1.2",
-        "paysera/lib-php-cs-fixer-config": "^2.3",
-        "phpunit/phpunit": "^6.5",
-        "symfony/yaml": "^3.4.34|^4.3"
+        "doctrine/annotations": "^1.14",
+        "doctrine/doctrine-bundle": "~2.6.0",
+        "doctrine/orm": "^2.10",
```
Comment (borilyordanov) [inline composer.json:34]: We will need to add some of Doctrine packages as non-dev one, because we have references to them in [FormatterTrait](src/Service/Formatter/FormatterTrait.php).

Actually we have other soft dependencies except Doctrine.

**Review by borilyordanov — CHANGES_REQUESTED** [review]
1. In [IterationEndListener](src/Listener/IterationEndListener.php) we are referring to `Sentry\FlushableClientInterface`, which was [removed](https://github.com/getsentry/sentry-php/blob/master/UPGRADE-3.0.md) from `sentry/sentry v3.x`.
2. In [CorrelationIdListener](src/Listener/CorrelationIdListener.php#L22) we are type-hinting `FilterResponseEvent`, which is [deprecated](https://github.com/symfony/symfony/blob/5.0/UPGRADE-5.0.md#httpkernel) from Symfony 4.3. Does it make sense to do some changes at the moment or we can postpone it until we add support for Symfony 5.x?

File: composer.json (hunk 6 above)
Comment (daniel-valchev-paysera) [reply composer.json:34]: There is no need to make Doctrine packages mandatory. FormatterTrait tries to deal with them if they are installed but they're not mandatory for it to work. The code first checks for instances of the Doctrine classes before actually using them.

The other soft dependencies are added.

**Comment by daniel-valchev-paysera** [comment]
> 2. In [CorrelationIdListener](src/Listener/CorrelationIdListener.php#L22) we are type-hinting `FilterResponseEvent`, which is [deprecated](https://github.com/symfony/symfony/blob/5.0/UPGRADE-5.0.md#httpkernel) from Symfony 4.3. Does it make sense to do some changes at the moment or we can postpone it until we add support for Symfony 5.x?

Current CorrelationIdListener cannot work on both Symfony 3.4 and Symfony 5.x as in 3.4 the replacement class doesn't exist yet. We'll have to do a workaround. Anyways, currently, we don't support Symfony 5.x. I suggest leaving it for a future update.

**Review by borilyordanov — APPROVED** [review]
Looks good to me.
